### ISSUE TITLE
Handling occasionally small turning back of trajectories

### ISSUE THREAD

**Issue by cyang-kth** [cyang-kth/fmm#98]
![image](https://user-images.githubusercontent.com/6810084/89098783-d6eebe80-d3ea-11ea-950a-d89d686efc1e.png)

A parameter can be added to the configurations of fmm.

**Comment by cyang-kth** [comment]
Add a parameter called reverse tolerance to the mm configuration.

**Comment by cyang-kth** [comment]
This issue is handled at this branch https://github.com/cyang-kth/fmm/tree/reverse-movement.

**Comment by windingwind** [comment]
> This issue is handled at this branch https://github.com/cyang-kth/fmm/tree/reverse-movement.

I have the same problem, thanks! And what should I do to solve this problem after switching to the reverse-movement branch? Should I add any config item in, like, STMATCHConfig?

**Comment by cyang-kth** [comment]
@windingwind 
Currently I only implemented this feature in fmm (not yet in stmatch)

The extra option is called `--reverse_tolerance` that is the reverse moving proportion allowed on an edge relative to its length.  

```bash
fmm --ubodt network/ubodt.txt --network network/edges.shp --network_id fid \
--source u --target v --gps gps/noisy_traj.csv --gps_id nid -k 8 -r 0.003 -e 0.0005 \
--output gps/mr.txt --reverse_tolerance 0.5
```

That branch is not completed yet. If you switch to that branch, I think the stmatch may not work properly as some common parts shared by fmm and stmatch are updated only for fmm.

**Comment by windingwind** [comment]
@cyang-kth Thanks! I'll try it.

**Comment by cyang-kth** [comment]
This issue has been solved by merging the reverse-movement branch 7c80e8ae256322bdac20ec019d2134af706af9c5.
